R/taxonomy.R

```diff
@@ -209,6 +219,25 @@ setMethod("checkTaxonomy", signature = c(x = "SummarizedExperiment"),
     }
 )
 
+#' @rdname taxonomy-methods
+#'@export
+# Function to set taxonomy ranks
+setTaxonomyRanks <- function(rank) {
+    rank <- tolower(rank)
+    # Check if rank is a character vector with length >= 1
+    if (!is.character(rank) || length(rank) < 1  || any(rank == "" | rank == " " | rank == "\t" | rank == "-" | rank == "_") || any(grepl("\\s{2,}", rank))) {
```

Comment:
I think it is enough to check that the vector is character and length is positive. Or does these give any addtional value?